text/0027-filter-and-facet-behavior.md

```diff
@@ -0,0 +1,205 @@
+
+- `<`, `=<`, `>`, `=>` => only operates on numbers. MeiliSearch returns only the documents with numbers for this field.
+ex: `price > 19` does not return `"price": "20"` but returns `"price": 20` -> no document conversions are done.
+- `=`, `!=`/`NOT` => string and number operation. MeiliSearch returns the documents with numbers, string (and array of strings) for this field.
+- cannot filter on `null`, objects, arrays of "undefined elements" (ex: array of `null`)
+
+#### Accepted syntaxes for `filters`
+
+Two syntaxes will be accepted for the `filters` parameter during the search:
+
+- the string syntax with the `AND`/`OR` operators
+ex:
+```json
+{
+    "filters": "(genres = Comedy OR genres = Romance) AND director = 'Mati Diop'"
+}
+```
+
+- the array syntax:
+ex:
+```json
+{
+    "filters": [["genres = Comedy", "genres = Romance"], "director = 'Mati Diop'"]
+}
+```
+
+
+#### `filters` and `facetsDistribution` behavior
+
+##### MeiliSearch v0.20.0 with `filters`
+
+In MeiliSeach v0.20.0, with the following documents
+
+```json
+[
+  { "id": 456,  "genre": "adventure", "price": 12  },
+  { "id": 1,    "genre": "fantasy",   "price": 456 }
+]
+```
+
+...and the following search
+
+```json
+{
+    "q": "",
+    "filters": "price = 12",
+    "facetsDistribution": ["genre"]
+}
+```
+
+...we get the following results:
+
+```json
+{
+    "hits": [{ "id": 456,  "genre": "adventure", "price": 12 }],
+    "facetsDistribution": {
+        "genre": {
+            "fantasy": 1,
+            "adventure": 1
+        }
+    }
+}
+```
+
+=> `fantasy` is set to `1` despite the fantasy book does not have a `price` equals to `12` as required in the `filters`.
+=> the `filters` and the `facetsDistribution` are not related: the `facetsDistribution` is applied before the filters.
+
+##### MeiliSearch v0.20.0 with `facetFilters`
+
+In MeiliSeach v0.20.0, with the following documents
+
+```json
+[
+  { "id": 456,  "genre": "adventure", "price": "12"  },
+  { "id": 1,    "genre": "fantasy",   "price": "456" }
+]
+```
+(the test is done with `price` as strings because MeiliSearch v0.20.0 cannot facet on numbers)
+
+...and the following search
+
+```json
+{
+    "q": "",
+    "facetFilters": ["price:12"],
+    "facetsDistribution": ["genre"]
+}
+```
+
+...we get the following results:
+
+```json
+{
+    "hits": [{ "id": 456,  "genre": "adventure", "price": "12" }],
+    "facetsDistribution": {
+        "genre": {
+            "fantasy": 0,
+            "adventure": 1
+        }
+    }
+}
+```
+
+=> `fantasy` is set to `0` because the fantasy book does not have a `price` equals to `12` as required in the `facetFilters`.
+=> the `filters` and the `facetsDistribution` are related: the `facetsDistribution` is applied after the facet filters.
+
+##### Final decision for v0.21.0
+
+MeiliSearch v0.21.0 will behave as the `facetFilters` already behave in MeiliSearch v0.20.0: the `facetsDistribution` will be applied after the filters.
+
+#### TLDR; all the breaking changes
```

Comment:
@bidoubiwa @react-learner Do we need more precise indications for the documentation impact? Thanks!

Reply:
Maybe juste some more information about the processing of the filters and facets. Is it done both at indexation time now? Or filters are still sorted at runtime?

Reply:
Could we also have an example of an extreme filtering query? With a lot of facets, and filter, with AND and OR's? Nested in parentheses, etc

Reply:
@bidoubiwa I've added some information. Hope it helps!